src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -31,23 +31,27 @@ public static CodeWriter AppendNullableValue(this CodeWriter writer, CSharpType
             return writer;
         }
 
-        public static void WriteParameter(this CodeWriter writer, Parameter clientParameter, bool includeDefaultValue = true)
+        public static void WriteParameter(this CodeWriter writer, Parameter clientParameter, bool enforceDefaultValue = false)
```

Comment:
Was nobody using the `includeDefaultValue` parameter?

Reply:
Nope. It was always default value, and `CodeWriterExtensions` is an internal type.

Reply:
Wonder if we should just create a new set of `Parameter`s

Reply:
`clientParameter.DefaultValue` must not be `null` to write default value, however `Constant.Default(CSharpType)` throws exception if type isn't nullable. So we either have to make non-nullable parameters required, or use a flag to override this behavior